src/server/analytic_services/prometheus_reporting.js

```diff
@@ -1,485 +1,111 @@
-        help: 'Object Savings Physical',
-    },
-    generate_default_set: true,
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'rebuild_progress',
-    configuration: {
-        name: get_metric_name('rebuild_progress'),
-        help: 'Rebuild Progress',
-    },
-    generate_default_set: true,
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'rebuild_time',
-    configuration: {
-        name: get_metric_name('rebuild_time'),
-        help: 'Rebuild Time',
-    },
-    generate_default_set: true,
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'bucket_healthy',
-    configuration: {
-        name: get_metric_name('bucket_status'),
-        help: 'Bucket Health',
-        labelNames: ['bucket_name'],
-    },
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'bucket_capacity_precent',
-    configuration: {
-        name: get_metric_name('bucket_capacity'),
-        help: 'Bucket Capacity Precent',
-        labelNames: ['bucket_name'],
-    },
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'bucket_quota_precent',
-    configuration: {
-        name: get_metric_name('bucket_quota'),
-        help: 'Bucket Quota Precent',
-        labelNames: ['bucket_name'],
-    },
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'resource_healthy',
-    configuration: {
-        name: get_metric_name('resource_status'),
-        help: 'Resource Health',
-        labelNames: ['resource_name'],
-    },
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'system_links',
-    configuration: {
-        name: get_metric_name('system_links'),
-        help: 'System Links',
-        labelNames: ['resources', 'buckets', 'dashboard'],
-    },
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'health_status',
-    configuration: {
-        name: get_metric_name('health_status'),
-        help: 'Health status',
-    },
-    generate_default_set: true,
-}]);
-
-
-class PrometheusReporting {
-    static instance() {
-        PrometheusReporting._instance = PrometheusReporting._instance || new PrometheusReporting();
-        return PrometheusReporting._instance;
-    }
+// Import the reports.
+const { is_enabled } = require('./prometheus_reports/base_prometheus_report');
+const { NodeJsReport } = require('./prometheus_reports/nodejs_report');
+const { NooBaaCoreReport } = require('./prometheus_reports/noobaa_core_report');
+const { NooBaaEndpointReport } = require('./prometheus_reports/noobaa_endpoint_report');
 
-    constructor() {
-        METRIC_RECORDS.filter(metric => metric.generate_default_set).forEach(metric => {
-            this[`set_${metric.metric_variable}`] = function(data) {
-                if (!this.enabled()) return;
-                this._metrics[metric.metric_variable].set(data);
-            };
-        });
-        this._prom_client = prom_client;
-        if (this.enabled()) {
-            this.define_metrics();
-        }
-    }
+const reports = new Map();
```

Comment:
I can understand the pros and cons of having them as a global state and not stored in a map.
On the plus side, I like that the code will be simpler, with no need for name strings of config values for the name.
On the minus side, it will make adding new reports much more verbose and troublesome because you need to be aware of each report individually. 

I think in this case the pros are greater than the cons, so I will do the change, but In any case, I don't see any value in adding a `Reports` class. 
My change will be to have, for each report a global variable and a `get_XXXX_report` function exposed in `prometheus_reporting.js`. For 3 reasons:
1. calling code will be simpler, instead of `prom_reporting.Reports.instance.get_XXXX_report()` it will be just `prom_reporting.get_X_report()`
2. `prom_reporting` is acting more like a utils collection at this point.
3 . Adding a class that is a singleton is kind of meaningless in the nodejs world where you have module scope which acts like a singleton. I can understand the need in other languages like C or C++ where you do not have the semantics of just running code in the global scope.

Reply:
Well, not only, singletons and classes are better for creating tests.
I don't care for singletons that prevent creation of more objects, only such that have the default instance for the entire system to easily locate it.
Using classes allows you to do better whitebox testing by replacing some functions in it to mock some behaviors.
When you'll get to testing this, you'll see :)

Reply:
I agree about white boxing on the reports themselves not utils.
> Using classes allows you to do better whitebox testing by replacing some functions in it to mock some behaviors.
When you'll get to testing this, you'll see ;)

In js the simplest thing to mock is a module function, you just replace the function on the exports objects

Reply:
Perhaps it is possible as you suggest, but why do you think it's easier to mock a module exports object which all the other code in the system refers to, rather than create a new unrelated instance that the other code cannot refer to. It's not critical, but I would encourage you to class a class to encapsulate state and functionality.